osf/management/commands/check_crossref_dois.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import requests
+
+from datetime import timedelta
+
+from framework.celery_tasks import app as celery_app
+from website import settings
+from website import mails
+from django.utils import timezone
+from django.core.management.base import BaseCommand
+
+import django
+django.setup()
+
+from osf.models import Preprint
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def check_crossref_dois(dry_run=True):
+    """
+    This script is to check for any DOI confirmation messages we may have missed during downtime and alert admins to any
+    DOIs that have been pending for X number of days.
+    :param dry_run:
+    :return:
+    """
+    # Create one enormous url to check if all our pending crossref DOIs are good, then set all identifers
+    preprints_with_pending_dois = Preprint.objects.filter(preprint_doi_created__isnull=True,
```

Comment:
Should this also include a "stuck" timedelta filter?

Reply:
I discussed with product, they don't have a problem with DOIs that have been "stuck" for an hour or however long getting set here. Seems extraneous, could potential cause problems if we have to change the time window for being "stuck".

Reply:
Maybe a better question would be "does it harm anything if we include preprints made 10 seconds before this script runs?"

If Crossref just ignores it, I suppose it doesn't matter much, but what if this fetches and sets it on our side before the email they send is delivered and parsed? Does that part have appropriate error handling if the preprint already has a DOI?

Reply:
Good point, I'll add this in